creusot/tests/should_succeed/linked_list.rs

```diff
@@ -0,0 +1,109 @@
+extern crate creusot_contracts;
+use creusot_contracts::{
+    logic::Seq,
+    ptr_own::{PtrOwn, RawPtr},
+    *,
+};
+
+struct Cell<T> {
+    v: T,
+    next: RawPtr<Cell<T>>,
+}
+
+pub struct List<T> {
+    // actual data
+    first: RawPtr<Cell<T>>,
+    last: RawPtr<Cell<T>>,
+    // ghost
+    seq: GhostBox<Seq<PtrOwn<Cell<T>>>>,
+}
+
+impl<T> Invariant for List<T> {
+    #[open]
+    #[predicate]
+    fn invariant(self) -> bool {
+        pearlite! {
+            (*self.seq == Seq::EMPTY &&
+             self.first.is_null_logic() &&
+             self.last.is_null_logic())
+            ||
+            (self.seq.len() > 0 &&
+             self.first == self.seq[0].ptr() &&
+             self.last  == self.seq[self.seq.len() - 1].ptr() &&
+             // the cells in `seq` are chained properly
+             (forall<i:Int> 0 <= i && i < self.seq.len() - 1 ==>
+                 self.seq[i].val().next == self.seq[i+1].ptr()) &&
+             self.seq[self.seq.len() - 1].val().next.is_null_logic())
+        }
+    }
+}
+
+impl<T> View for List<T> {
+    type ViewTy = Seq<T>;
+
+    #[logic]
+    #[open(self)]
+    fn view(self) -> Self::ViewTy {
+        pearlite! {
+            seq_map(*self.seq, |ptr_own: PtrOwn<Cell<T>>| ptr_own.val().v)
+        }
+    }
+}
+
+// TODO: allow "- 1" on Ints in ghost code directly
```

Comment:
maybe we should have an issue to track this TODO…

Reply:
I opened #1251 to track this.